geode-core/src/main/java/org/apache/geode/distributed/internal/streaming/StreamingOperation.java

```diff
@@ -501,15 +517,23 @@ public void fromData(DataInput in) throws IOException, ClassNotFoundException {
       this.pdxReadSerialized = in.readBoolean();
       Version senderVersion = InternalDataSerializer.getVersionForDataStream(in);
       boolean isSenderAbove_8_1 = senderVersion.compareTo(Version.GFE_81) > 0;
+      InternalCache cache = null;
+      Boolean initialPdxReadSerialized = false;
+      try {
+        cache = (InternalCache) CacheFactory.getAnyInstance();
```

Comment:
Use GemFireCacheImpl.getForPdx. For two reasons - CacheFactory.getAnyInstance gets locks, so this could cause a deadlock. And getForPdx works while the cache is closing.

Reply:
done